crypto/ec/ec_ameth.c

```diff
@@ -568,6 +570,131 @@ static int ec_pkey_param_check(const EVP_PKEY *pkey)
     return EC_GROUP_check(eckey->group, NULL);
 }
 
+static
+size_t ec_pkey_dirty_cnt(const EVP_PKEY *pkey)
+{
+    return pkey->pkey.ec->dirty_cnt;
+}
+
+static ossl_inline
+int ecparams_to_params(const EC_GROUP *ecg, OSSL_PARAM_BLD *tmpl)
+{
+    int curve_nid;
+
+    if (ecg == NULL)
+        return 0;
+
+    curve_nid = EC_GROUP_get_curve_name(ecg);
+
+    if (curve_nid == NID_undef) {
+        /* explicit parameters */
+
+        /*
+         * TODO(3.0): should we support explicit parameters curves?
+         */
+        return 0;
+    } else {
+        /* named curve */
+        const char *curve_name = NULL;
+
+        if ((curve_name = OBJ_nid2sn(curve_nid)) == NULL)
+            return 0;
+
+        if (!ossl_param_bld_push_utf8_string(tmpl, OSSL_PKEY_PARAM_EC_NAME, curve_name, 0))
+            return 0;
+    }
+
+    return 1;
+}
+
+static
+void *ec_pkey_export_to(const EVP_PKEY *pk, EVP_KEYMGMT *keymgmt,
+                        int want_domainparams)
```

Comment:
You can drop `want_domainparams` entirely.  This function is assumed to export *everything* there is in the legacy key, so what actually ends up on the provider side is entirely data driven.